evap/results/views.py

```diff
@@ -339,3 +329,71 @@ def add_warnings(evaluation, evaluation_result):
+
+
+def evaluation_detail_parse_get_parameters(request, evaluation):
+    if not evaluation.can_results_page_be_seen_by(request.user):
+        raise PermissionDenied
+
+    if request.user.is_reviewer:
+        view = request.GET.get('view', 'public')  # if parameter is not given, show public view.
+    else:
+        view = request.GET.get('view', 'full')  # if parameter is not given, show own view.
+    if view not in ['public', 'full', 'export']:
+        view = 'public'
+
+    view_as_user = request.user
+    contributor_id = int(request.GET.get('contributor_id', request.user.id))
+    if view == 'export' and request.user.is_staff:
+        view_as_user = UserProfile.objects.get(id=contributor_id)
+    contributor_id = contributor_id if contributor_id != request.user.id else None
+
+    represented_users = [view_as_user]
+    if view != 'export':
+        represented_users += list(view_as_user.represented_users.all())
+    # redirect to non-public view if there is none because the results have not been published
+    if not evaluation.can_publish_rating_results and view == 'public':
+        view = 'full'
+
+    return view, view_as_user, represented_users, contributor_id
+
+
+def extract_evaluation_answer_data(request, evaluation):
+    # TextAnswerExcelExporter wants a dict from Question to tuple of contributor_name and string list (of the answers)
+
+    view, view_as_user, represented_users, contributor_id = evaluation_detail_parse_get_parameters(request, evaluation)
+
+    evaluation_result = collect_results(evaluation)
+    filter_text_answers(evaluation_result)
+    remove_textanswers_that_the_user_must_not_see(evaluation_result, view_as_user, represented_users, view)
+
+    results = dict()
+    for contribution_result in evaluation_result.contribution_results:
+        contributor_name = contribution_result.contributor.full_name if contribution_result.contributor is not None else ""
+        for questionnaire_result in contribution_result.questionnaire_results:
+            for result in questionnaire_result.question_results:
+                results[result.question] = (contributor_name, [answer.answer for answer in result.answers])
+
+    return results, contributor_id
+
+
+@reviewer_required
+def evaluation_text_answers_export(request, evaluation_id):
+    evaluation = get_object_or_404(Evaluation, id=evaluation_id)
+
+    results, contributor_id = extract_evaluation_answer_data(request, evaluation)
+    contributor_name = UserProfile.objects.get(id=contributor_id).full_name if contributor_id is not None else None
+
+    filename = "Evaluation-Text-Answers-{}-{}-{}.xls".format(
+        evaluation.course.semester.short_name,
+        evaluation.full_name,
+        translation.get_language()
+    )
+
+    response = HttpResponse(content_type="application/vnd.ms-excel")
+    response["Content-Disposition"] = bytes("attachment; filename=\"{}\"".format(filename), "utf-8")
```

Comment:
I think this is smelly, especially if we can't really explain the behaviour with the old code or why the new code is any better. I'll try to find out why your problem occurred and whether this is a proper solution some time this week.

(I read your comment)

Reply:
Alrighty, so the issue is that [RFC 2183](https://tools.ietf.org/html/rfc2183), which defines the content-disposition header, only allows for us-ascii encoded strings in the header. To comply to this, django will format it the way it did for you if it contains non-ascii characters. Manually encoding it to a bytestring before will circumvent django's logic, but we're relying on non-standard behavior in the browsers then.

Django handles it [like this](https://github.com/django/django/blob/master/django/http/response.py#L445) in their FileResponse - sadly, this class is not compatible with having a byte stream in memory, it only supports real files in the file system.
```python
try:
     filename.encode('ascii')
     file_expr = 'filename="{}"'.format(filename)
except UnicodeEncodeError:
     file_expr = "filename*=utf-8''{}".format(quote(filename))

self['Content-Disposition'] = '{}; {}'.format(disposition, file_expr))
```
as defined in RFC 6266. [All modern browsers should support this](http://test.greenbytes.de/tech/tc2231/#attwithfn2231utf8). Thus, I'd say we should also do it like this.

How often do we currently have the case that we manually set the content disposition? Could you put this code snipped into a method (`make_file_download_response` or whatever) and use it in these places? That would be really nice.

Reply:
The pattern
```python
response = HttpResponse(content_type=...)
response["Content-Disposition"] = ...
```
is now handled by evaluation.tools.FileResponse (do we use Django's FileResponse somewhere?):
```python
response = FileResponse(filename, content_type=...)
```
This sets the header according to the code above.

I replaced this everywhere I found the word "Content-Disposition".